File: 1013-fibonacci-number/fibonacci-number.java
class Solution {
    public int fib(int n) {
        if(n<=0)
          return n;
        int i=1;
        int j=1;
        for(int k=3;k<=n;k++){
            int t=i+j;
            i=j;
            j=t;
        }
        return j;
    }
}

File: 1013-fibonacci-number/fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
     if n <= 0:
        return 0
     if n == 1 or n == 2:
        return 1

     i, j = 1, 1
     for _ in range(2, n):
        t = i + j
        i = j
        j = t

     return j